java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -462,7 +463,27 @@ private void executeMethodInvocation(MethodInvocationTree mit) {
     ProgramState.Pop unstack = programState.unstackValue(mit.arguments().size() + 1);
     programState = unstack.state;
     logState(mit);
-    programState = programState.stackValue(constraintManager.createMethodSymbolicValue(mit, unstack.values));
+    final SymbolicValue resultValue = constraintManager.createMethodSymbolicValue(mit, unstack.values);
+    programState = programState.stackValue(resultValue);
+    if (isNonNullMethod(mit)) {
+      programState = programState.addConstraint(resultValue, ObjectConstraint.NOT_NULL);
+    }
+  }
+
+  private boolean isNonNullMethod(MethodInvocationTree mit) {
+    IdentifierTree identifier;
+    if (mit.methodSelect().is(Tree.Kind.IDENTIFIER)) {
+      identifier = (IdentifierTree) mit.methodSelect();
+    } else if (mit.methodSelect().is(Tree.Kind.MEMBER_SELECT)) {
+      identifier = ((MemberSelectExpressionTree) mit.methodSelect()).identifier();
+    } else {
+      return false;
+    }
+    if (identifier.symbol() instanceof JavaSymbol) {
```

Comment:
You want to check if the method is _not unknown_ (`!symbol.isUnknown()`), not if it's a JavaSymbol!